modules/caption-impl/src/main/java/org/opencastproject/caption/converters/WebVttCaptionConverter.java

```diff
@@ -107,7 +122,28 @@ public String getExtension() {
 
   @Override
   public Type getElementType() {
-    return MediaPackageElement.Type.Attachment;
+    return mediapackageElementType;
+  }
+
+  /**
+   * Converts the configured property 'mediapackage-element-type' into the corresponding enum 'MediapackageElement.Type'
+   *
+   * @param cc ComponentContext
+   * @return the configured mediapackage element type
+   */
+  private MediaPackageElement.Type getConfiguredMediapackageElementType(ComponentContext cc) {
+    String mediapackageElementType = null;
+    try {
+      mediapackageElementType = OsgiUtil.getOptCfg(cc.getProperties(), MEDIAPACKAGE_ELEMENT_TYPE_CONFIG_KEY).get();
+      // capitalize the first letter
+      mediapackageElementType = mediapackageElementType.substring(0, 1).toUpperCase()
+          + mediapackageElementType.substring(1);
+      return MediaPackageElement.Type.valueOf(mediapackageElementType);
+    } catch (Exception e) {
+      logger.warn("Couldn't convert configuration '{}'='{}' into enum. Using default '{}'.",
+          MEDIAPACKAGE_ELEMENT_TYPE_CONFIG_KEY, mediapackageElementType, DEFAULT_MEDIAPACKAGE_ELEMENT_TYPE);
+      return DEFAULT_MEDIAPACKAGE_ELEMENT_TYPE;
```

Comment:
Please don't do that!
As an administrator, if I accidentally misconfigure this, don't just choose something.
Use the default __only__ if nothing is set.
If I set something that does not make any sense, please fail, so I realize that immediately and don't realize after a few weeks that Opencast did always silently do the wrong thing.

Reply:
Sorry for that. Now it will fail if something wrong was configured and a detailed message will be printed.